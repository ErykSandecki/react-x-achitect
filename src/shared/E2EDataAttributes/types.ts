// types
import { E2EAttribute } from 'types/e2e';

export type E2EType = E2EAttribute;
export type E2EValue = number | string;
