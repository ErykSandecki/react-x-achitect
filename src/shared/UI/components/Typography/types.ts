// types
import { TTypographyProps } from './Typography';

export type TOmittedTypographyProps = Omit<TTypographyProps, 'fontType'>;
