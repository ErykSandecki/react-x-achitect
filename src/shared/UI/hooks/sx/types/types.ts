// types
import { TSXSpacings } from './spacings';

export type TSX = {
  spacings?: TSXSpacings;
};
