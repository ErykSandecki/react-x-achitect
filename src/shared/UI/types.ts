// types
import { TObject } from 'types';
import { TSX } from './hooks/sx/types/types';

export type TClasses = TObject<string>;

export type TUIProps<T extends TClasses, K = TSX> = {
  classes?: T;
  sx?: K;
};
