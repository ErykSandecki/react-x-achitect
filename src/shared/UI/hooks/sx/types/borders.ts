// types
import { Border, BorderColor } from '../enums/borders';
import { TObject } from 'types';
import { TSXColor } from './types';

export type TSXBorders = TObject<number, typeof Border, 'optional'> &
  TObject<TSXColor, typeof BorderColor, 'optional'> & {
    borderRadius?: number | string;
  };
