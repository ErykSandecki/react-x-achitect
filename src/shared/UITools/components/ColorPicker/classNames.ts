export const className = 'ColorPicker';

export const classes = {
  className,
};
