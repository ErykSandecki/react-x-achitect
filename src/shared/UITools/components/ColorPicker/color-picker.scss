.ColorPicker {
  &__hex-input {
    width: 100%;
  }

  &__picker {
    border-radius: 2px;
    height: 14px;
    width: 14px;
  }
}
