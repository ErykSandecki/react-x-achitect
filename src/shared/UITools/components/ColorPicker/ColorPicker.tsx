import { ColorPicker as ColorPickerAntd, ColorPickerProps } from 'antd';
import { FC, useRef, useState } from 'react';

// components
import Color from '../Color/Color';
import FieldGroup from '../FieldGroup/FieldGroup';
import Panel from './components/Panel/Panel';
import TextField from '../TextField/TextField';

// hooks
import { useColorPickerEvents } from './hooks/useColorPickerEvents';
import { useTheme } from 'hooks';

// others
import { classes, className, classNames } from './classNames';

// styles
import styles from './color-picker.scss';

// types
import { E2EAttribute, KeyboardKeys, TBackground } from 'types';
import { TE2EDataAttributeProps } from '../../../E2EDataAttributes/E2EDataAttribute';
import { TUIProps } from '../../../UI/types';

// utils
import { getAttributes } from '../../../E2EDataAttributes/utils';
import { handleSubmitInput, hexToRgb } from 'utils';

export type TColorPickerProps = TUIProps<typeof classes> &
  Omit<ColorPickerProps, 'arrow' | 'onOpenChange' | 'open' | 'panelRender'> & {
    alpha: TBackground['alpha'];
    color: string;
    e2eValue?: TE2EDataAttributeProps['value'];
    onChangeAlpha: (value: string) => void;
    onChangeColor: (alpha: string, value: string) => void;
  };

export const ColorPicker: FC<TColorPickerProps> = ({
  alpha,
  classes = { className: '' },
  color,
  e2eValue = '',
  onChangeAlpha: onChangeAlphaHandler,
  onChangeColor: onChangeColorHandler,
  ...restProps
}) => {
  const { classNamesWithTheme, cx } = useTheme(classNames, styles);
  const {
    alphaValue,
    colorValue,
    onBlurAlpha,
    onBlurColor,
    onChangeAlpha,
    onChangeColor,
    onChangeColorPicker,
  } = useColorPickerEvents(
    alpha,
    color,
    onChangeAlphaHandler,
    onChangeColorHandler,
  );
  const inputAlpha = useRef(null);
  const inputColor = useRef(null);
  const [visible, setVisible] = useState(false);

  return (
    <FieldGroup>
      <TextField
        className={cx(classNamesWithTheme.hexInput)}
        onBlur={onBlurColor}
        onChange={onChangeColor}
        onKeyDown={(event) =>
          handleSubmitInput(KeyboardKeys.enter, inputColor.current)(event)
        }
        ref={inputColor}
        startAdornment={
          <ColorPickerAntd
            arrow={false}
            className={cx(classes.className, classNamesWithTheme[className])}
            onChange={onChangeColorPicker}
            onOpenChange={(visible) => setVisible(visible)}
            open={visible}
            panelRender={(children) => (
              <Panel setVisible={setVisible}>{children}</Panel>
            )}
            value={hexToRgb(colorValue, parseInt(alphaValue))}
            {...getAttributes(E2EAttribute.colorPicker, e2eValue)}
            {...restProps}
          >
            <div>
              <Color alpha={alpha} color={color} />
            </div>
          </ColorPickerAntd>
        }
        value={colorValue.replace('#', '')}
      />
      <TextField
        className={cx(classNamesWithTheme.alphaInput)}
        endAdorment={<div>%</div>}
        max={100}
        min={0}
        onBlur={onBlurAlpha}
        onChange={onChangeAlpha}
        onKeyDown={(event) =>
          handleSubmitInput(KeyboardKeys.enter, inputAlpha.current)(event)
        }
        ref={inputAlpha}
        type="number"
        value={alphaValue}
      />
    </FieldGroup>
  );
};

export default ColorPicker;
