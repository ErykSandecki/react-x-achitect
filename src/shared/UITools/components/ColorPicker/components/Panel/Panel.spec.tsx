import { render } from '@testing-library/react';

// components
import ColorPicker from './Panel';

// types
import { E2EAttribute } from 'types/e2e';

// utils
import { getByE2EAttribute } from 'test/testHelpers';

const className = 'className';

describe('Typography props', () => {});

describe('Box snapshots', () => {});
