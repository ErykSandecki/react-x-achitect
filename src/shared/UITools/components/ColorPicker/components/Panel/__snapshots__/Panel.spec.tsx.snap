// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel snapshots should render Panel 1`] = `
<DocumentFragment>
  <div
    class="Panel Panel--light"
    data-test-color-picker-panel=""
  >
    <div
      class="Panel__header Panel__header--light"
    >
      <div
        class="css-q0izro Tabs Tabs--light"
        data-test-box=""
      >
        <div
          class="Tabs__tab Tabs__tab--light Tabs__tab--active Tabs__tab--active--light Tabs__tab--disabled Tabs__tab--disabled--light"
          data-test-tab="custom"
        >
          Custom
        </div>
      </div>
      <div
        class="css-q0izro Icon Icon--light Icon--clickable Icon--clickable--light"
        data-test-icon="Close"
        height="11"
        name="Close"
        width="11"
      />
    </div>
    <div
      class="ant-color-picker-slider-container"
    />
    <div
      class="Panel__sample Panel__sample--light"
    >
      <div
        class="css-q0izro ButtonIcon ButtonIcon--light"
        data-test-button-icon=""
      >
        <div
          class="css-q0izro Icon Icon--light Icon--neutral1 Icon--neutral1--light"
          data-test-icon="Sample"
          height="14"
          name="Sample"
          width="14"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
